Source/WebInspectorUI/UserInterface/Views/Layers3DContentView.js

```diff
@@ -417,6 +518,64 @@ WI.Layers3DContentView = class Layers3DContentView extends WI.ContentView
         WI.layerTreeManager.showPaintRects = !WI.layerTreeManager.showPaintRects;
     }
 
+    _handleLayerContentsSettingChanged(event)
+    {
+        this._refreshLayersButtonNavigationItem.enabled = WI.settings.layers3DShowLayerContents.value;
+
+        for (let loadContext of this._pendingTextureLoads.values())
+            loadContext.cancelled = true;
+        this._pendingTextureLoads.clear();
+
+        this._refreshAllLayers();
+    }
+
+    _handleRefreshLayersButtonClicked(event)
+    {
+        for (let loadContext of this._pendingTextureLoads.values())
+            loadContext.cancelled = true;
+        this._pendingTextureLoads.clear();
+
+        this._refreshAllLayers();
+    }
+
+    _refreshAllLayers()
+    {
+        for (let [layerId, layerGroup] of this._layerGroupsById) {
+            let layer = layerGroup.userData.layer;
+            for (let child of layerGroup.children) {
+                child.geometry?.dispose();
+                child.material?.map?.dispose();
+                child.material?.dispose();
+            }
+            layerGroup.clear();
+
+            let fillMesh = this._createLayerMesh(layer.bounds);
```

Comment:
i think the rest of this function is the same as what's done in `_createLayerGroup`.  could we create a separate function like `_populateLayerGroup` to avoid the repeated code?